test_util/azure.py

```diff
@@ -35,77 +35,123 @@ def validate_hostname_prefix(prefix):
     assert re.match('^[a-z][a-z0-9-]{1,61}[a-z0-9]$', prefix), 'Invalid DNS prefix: {}'.format(prefix)
 
 
+def check_json_object(obj):
```

Comment:
I don't think you are seeing the intention of this code. This is to fill in the cast map. It doesn't make any sense to try to cast something to an object or array when we are already using JSON/YAML.